From 1343d9acbee07ae045df1cad5f6836a3b7d02521 Mon Sep 17 00:00:00 2001
Message-Id: <1343d9acbee07ae045df1cad5f6836a3b7d02521.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Arun R Murthy <arun.r.murthy@intel.com>
Date: Mon, 13 Oct 2014 15:28:42 +0530
Subject: [PATCH 61/75] MUST_REBASE [VPG]: adf/intel/vlv: program the pfit
 credits

Program the pfi credits to 15 credits and also enable PFI credits to
resend to SSA.

Magic constants need need update, but as of now they are recommended
values by SV team

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: I47d42d997949de5d86db098f540ed9a9c0473461
Signed-off-by: Arun R Murthy <arun.r.murthy@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_mode_set.c    |    5 +++++
 .../video/adf/intel/include/core/vlv/vlv_dc_regs.h |    2 ++
 2 files changed, 7 insertions(+)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_mode_set.c b/drivers/video/adf/intel/core/vlv/vlv_mode_set.c
index f94459b..2aa4c0e 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_mode_set.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_mode_set.c
@@ -168,6 +168,11 @@ int vlv_display_on(struct intel_pipe *pipe)
 	vlv_program_wm();
 	/* Trickle feed is disabled by default */
 	REG_WRITE(MI_ARB_VLV, 0x00);
+	/* program the pfi credits, first disable and then program */
+	if (REG_READ(GCI_CONTROL) != 0x78004000) {
+		REG_WRITE(GCI_CONTROL, 0x00004000);
+		REG_WRITE(GCI_CONTROL, 0x78004000);
+	}
 
 	return 0;
 }
diff --git a/drivers/video/adf/intel/include/core/vlv/vlv_dc_regs.h b/drivers/video/adf/intel/include/core/vlv/vlv_dc_regs.h
index 8bfe19b..8de921a 100644
--- a/drivers/video/adf/intel/include/core/vlv/vlv_dc_regs.h
+++ b/drivers/video/adf/intel/include/core/vlv/vlv_dc_regs.h
@@ -1470,6 +1470,8 @@
 #define   CZCLK_FREQ_MASK	0xf
 #define GMBUSFREQ_VLV		(VLV_DISPLAY_BASE + 0x6510)
 
+#define GCI_CONTROL		(VLV_DISPLAY_BASE + 0x650c)
+
 /*
  * Palette regs
  */
-- 
1.7.9.5

